client/asset/eth/rpcclient.go

```diff
@@ -0,0 +1,215 @@
+}
+
+// transactionReceipt uses a raw request to retrieve a transaction's receipt.
+func (c *rpcclient) transactionReceipt(ctx context.Context, txHash common.Hash) (*types.Receipt, error) {
+	res := new(types.Receipt)
+	if err := c.c.CallContext(ctx, res, "eth_getTransactionReceipt", txHash.String()); err != nil {
+		return nil, err
+	}
+	return res, nil
+}
+
+// pendingTransactions returns pending transactions.
+func (c *rpcclient) pendingTransactions(ctx context.Context) ([]*types.Transaction, error) {
+	var ptxs []*types.Transaction
+	err := c.c.CallContext(ctx, &ptxs, "eth_pendingTransactions")
+	if err != nil {
+		return nil, err
+	}
+	return ptxs, nil
+}
+
+// addPeer adds a peer.
+func (c *rpcclient) addPeer(ctx context.Context, peer string) error {
+	return c.c.CallContext(ctx, nil, "admin_addPeer", peer)
+}
+
+// blockNumber gets the block number at time of calling.
+func (c *rpcclient) blockNumber(ctx context.Context) (uint64, error) {
+	bn, err := c.ec.BlockNumber(ctx)
+	if err != nil {
+		return 0, err
+	}
+	return bn, nil
+}
+
+// nodeInfo retrieves useful information about a node.
+func (c *rpcclient) nodeInfo(ctx context.Context) (*p2p.NodeInfo, error) {
+	info := new(p2p.NodeInfo)
+	if err := c.c.CallContext(ctx, info, "admin_nodeInfo"); err != nil {
+		return nil, err
+	}
+	return info, nil
+}
+
+// listWallets list all of the wallet's wallets? and accounts along with details
+// such as locked status.
+func (c *rpcclient) listWallets(ctx context.Context) ([]rawWallet, error) {
+	var res []rawWallet
+	if err := c.c.CallContext(ctx, &res, "personal_listWallets"); err != nil {
+		return nil, err
+	}
+	return res, nil
+}
+
+// sendTransaction uses a raw request to send tx.
+func (c *rpcclient) sendTransaction(ctx context.Context, tx map[string]string) (common.Hash, error) {
+	res := common.Hash{}
+	err := c.c.CallContext(ctx, &res, "eth_sendTransaction", tx)
+	if err != nil {
+		return common.Hash{}, err
+	}
+	return res, nil
+}
+
+// syncProgress return the current sync progress. Returns no error and nil when not syncing.
+func (c *rpcclient) syncProgress(ctx context.Context) (*ethereum.SyncProgress, error) {
+	return c.ec.SyncProgress(ctx)
+}
+
+// importAccount imports an account into the ethereum wallet by private key
+// that can be unlocked with password.
+func (c *rpcclient) importAccount(pw string, privKeyB []byte) (*accounts.Account, error) {
+	privKey, err := crypto.ToECDSA(privKeyB)
+	if err != nil {
+		return new(accounts.Account), fmt.Errorf("error parsing private key: %v", err)
+	}
+	ks := c.n.AccountManager().Backends(keystore.KeyStoreType)[0].(*keystore.KeyStore)
+	acct, err := ks.ImportECDSA(privKey, pw)
+	if err != nil {
+		return nil, err
+	}
+	return &acct, nil
+}
+
+// wallet returns a wallet that owns acct from an ethereum wallet.
+// accounts.Wallet is an interface.
+func (c *rpcclient) wallet(acct accounts.Account) (accounts.Wallet, error) {
```

Comment:
Plans for this method (currently unused)?

Reply:
Removing for now... I must have been using it for something though.